src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -108,26 +111,24 @@ private void sequenceNumberSeen(Optional<SequenceNumber> seq) {
             .doOnNext(p -> sequenceNumberSeen(p.s()))
             .share();
 
-    Completable heartbeat =
-        ps.filter(Payload.isOp(OpCode.HELLO)).flatMapCompletable(p -> heartbeat(ws, p));
-
-    Completable identify =
+    disposables.add(
         ps.filter(Payload.isOp(OpCode.HELLO))
-            .flatMapCompletable(
-                p -> isResumable() ? resume(ws, token) : identify(ws, token, shard));
+            .flatMapCompletable(p -> heartbeat(ws, p))
+            .subscribe());
 
-    Observable<Event> events = ps.flatMapMaybe(payloads::toEvent).share();
+    disposables.add(
+        ps.filter(Payload.isOp(OpCode.HELLO))
+            .flatMapCompletable(p -> isResumable() ? resume(ws, token) : identify(ws, token, shard))
+            .subscribe());
```

Comment:
Could we somehow merge the 2 filtering methods? They're both filtering out the `OpCode.HELLO`

Reply:
Maybe, but then we'd have to do one of the following:
* Add another share, so that we can add multiple flows for `ps.filter(Payload.isOp(OpCode.HELLO)`
* Add a method like `onHello(ps)`, which doesn't really gain us much readability, but it does add another method.

Reply:
one and two both keep some duplication in the form of the heartbeat merge being in multiple places. The last one removes the duplication, but I'm not sure how much more readable it is.

For now, I'd like to keep them separate because it's not too much duplication, and we may end up wanting more logic on the heartbeat here (for example, if we received another Hello now I don't think we stop the first heartbeat, meaning we'd be having two heartbeats being sent)

Reply:
Alright, it's not a huge issue, just nitpicking. We can keep them separate especially if you have more plans for logic.